### PR TITLE
Add stop job action

### PR DESCRIPTION
**Pull request by cshintov**
Adds Stop and Purge

Issue #120

### CODE REVIEW

File: internal/tui/components/app/app.go [hunk 1]
```diff
@@ -211,8 +211,9 @@ func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 				}
 			case nomad.ExecPage:
 				m.getCurrentPageModel().SetInputPrefix("Enter command: ")
-			case nomad.AllocAdminConfirmPage:
-				// always make user go down one to confirm
+			case nomad.AllocAdminConfirmPage, nomad.JobAdminConfirmPage, nomad.JobAdminPage,  nomad.AllocAdminPage:
+				// always start at the top of the list for admin pages
+				// avoids surprises, consistently focusing on the first item
```
Comment (robinovitch61) [inline internal/tui/components/app/app.go:216]: I don't think we want this for the job admin page or alloc admin page - if someone is performing the same admin task across many entities, they'll want the focus on the same one each time similar to other pages

Related: https://github.com/robinovitch61/wander/issues/132

File: internal/tui/components/app/app.go [hunk 2]
```diff
@@ -337,6 +338,24 @@ func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 		newToast := toast.New(toastMsg)
 		m.getCurrentPageModel().SetToast(newToast, toastStyle)
 		cmds = append(cmds, tea.Tick(newToast.Timeout, func(t time.Time) tea.Msg { return toast.TimeoutMsg{ID: newToast.ID} }))
+
+	case nomad.JobAdminActionCompleteMsg:
+		toastMsg := fmt.Sprintf(
+			"%s completed successfully",
```
Comment (robinovitch61) [inline internal/tui/components/app/app.go:344]: Seeing some weird styling here 
image

File: internal/tui/components/app/app.go [hunk 3]
```diff
@@ -758,6 +802,36 @@ func (m Model) getCurrentPageCmd() tea.Cmd {
 				},
 			}
 		}
+	case nomad.JobAdminPage:
+		return func() tea.Msg {
+			// this does no async work, just constructs the task admin menu
```
Comment (robinovitch61) [inline internal/tui/components/app/app.go:807]:
```suggestion
			// this does no async work, just constructs the job admin menu
```

File: internal/tui/nomad/jobadmin.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+func GetJobAdminText(adminAction AdminAction, jobID string) string {
+	switch adminAction {
+	case StopJobAction, StopAndPurgeJobAction:
+		return fmt.Sprintf(
+			"%s job %s",
+			JobAdminActions[adminAction],
+			style.Bold.Render(jobID))
+	default:
+		return "----"
```
Comment (robinovitch61) [inline internal/tui/nomad/jobadmin.go:32]:
```suggestion
		return ""
```

For consistency with allocadmin

File: internal/tui/nomad/pages.go [hunk 5]
```diff
@@ -190,7 +204,7 @@ func (p Page) ShowsTasks() bool {
 }
 
 func (p Page) HasAdminMenu() bool {
-	adminMenuPages := []Page{AllTasksPage, JobTasksPage}
+	adminMenuPages := []Page{AllTasksPage, JobTasksPage, JobsPage}
```
Comment (robinovitch61) [inline internal/tui/nomad/pages.go:207]: 🥳

File: internal/tui/nomad/util.go [hunk 6]
```diff
@@ -22,9 +22,10 @@ const (
 	StopAllocAction
 	RestartJobAction
 	StopJobAction
+	StopAndPurgeJobAction
 )
 
-// AdminActionToKey and KeyToAdminAction are used to render the AllocAdminPage
+// AdminActionToKey and KeyToAdminAction are used to render the AdminPage
```
Comment (robinovitch61) [inline internal/tui/nomad/util.go:28]:
```suggestion
// AdminActionToKey and KeyToAdminAction are used to render the admin page menus
```

Comment (robinovitch61) [reply internal/tui/nomad/util.go:28]: (feel free to ignore this, nitpicky)

**Review by robinovitch61 — COMMENTED** [review]
Looks pretty good! Left a few comments

File: internal/tui/nomad/pages.go (hunk 5 above)
Comment (cshintov) [reply internal/tui/nomad/pages.go:207]: 🎉🕺

**Comment by cshintov** [comment]
Made the changes, and fixed a bug too, without the namespace the deregister(stop) call won't work.

So it was only working when started with `-n default`! Now it works even without it.

**Comment by robinovitch61** [comment]
NBD, but it helps me review easier if you push additional commits after reviews rather than force push :)

**Review by robinovitch61 — APPROVED** [review]
Nice!

**Comment by cshintov** [comment]
> NBD, but it helps me review easier if you push additional commits after reviews rather than force push :)

Will keep that in mind next time. Btw, what's NBD?

**Comment by robinovitch61** [comment]
No Big Deal, or "doesn't matter so much" :)

**Pull request merged**